cpu/stm32/periph/flashpage.c

```diff
@@ -130,9 +149,26 @@ static void _erase_page(void *page_addr)
     /* wait as long as device is busy */
     _wait_for_pending_operations();
 
-    /* reset PER bit */
+#ifdef FLASH_ACR_DCEN /* Flush the data cache after page erase */
+    if (FLASH->ACR & FLASH_ACR_DCEN) {
+        FLASH->ACR &= ~FLASH_ACR_DCEN;
+        FLASH->ACR |= FLASH_ACR_DCRST;
+        FLASH->ACR |= FLASH_ACR_DCEN;
+    }
+#endif
+#ifdef FLASH_ACR_ICEN /* Flush the instruction cache after page erase */
+    if (FLASH->ACR & FLASH_ACR_ICEN) {
+        FLASH->ACR &= ~FLASH_ACR_ICEN;
+        FLASH->ACR |= FLASH_ACR_ICRST;
+        FLASH->ACR |= FLASH_ACR_ICEN;
+    }
+#endif
+
+#ifdef FLASH_CR_PNB
+    /* reset PER bit (if the register settings exist) */
     DEBUG("[flashpage] erase: resetting the page erase bit\n");
-    CNTRL_REG &= ~(FLASH_CR_PER);
+    CNTRL_REG &= ~(FLASH_CR_PER | FLASH_CR_PNB);
+#endif
```

Comment:
This fixes the issue

```
#ifdef FLASH_CR_PNB
    /* reset PER bit (if the register settings exist) */
    DEBUG("[flashpage] erase: resetting the page erase bit\n");
    CNTRL_REG &= ~(FLASH_CR_PER | FLASH_CR_PNB);
#else
      CNTRL_REG &= ~(FLASH_CR_PER);
#endif
```

Reply:
@bergzand since you have been looking at this, doe that register exist on all smt32f2/4/7?